src/movies.js

```diff
@@ -1,16 +1,170 @@
+    let dramaMovies = array.filter((movie) => {
+        return movie.genre.includes("Drama") && movie.director === "Steven Spielberg";
+        // return movieType.genre.includes("drama");
+    })
+    return dramaMovies.length
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+const ratesAverage = (array) => {
+    
+    if (array != "") {
+
+        let averageRates = array.reduce((acc, i) => {
+            if (i.rate) {
+            return acc + i.rate / array.length
+            } else {
+                return acc + 0
+            }
+        }, 0)
+        //let avgRounded = averageRates.toFixed(2);
+        //return averageRates.toFixed(2);
+        
+        //return averageRates
+        let roundedAverage = (averageRates.toFixed(2));
+        roundedAverage = Number(roundedAverage); 
+        return roundedAverage
+
+    } else {
+        return 0; 
+    }
+    }
+
+
+
+
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+const dramaMoviesRate = (array) => {
+    if (array.genre !== "Drama") {
+        return 0;
+    }; 
+
+    let dramaMovies = array.filter((movie) => {
+        return movie.genre.includes("Drama");
+     });
+
+    let dramaAverageRates = array.reduce((acc, i) => {
+            return acc + i.rate / dramaMovies.length
+        }, 0)
+
+    return dramaAverageRates
+}
+
+    //
+
+//     let dramaMovies = array.filter((movie) => {
+//         return movie.genre.includes("Drama");
+//     });
+    
+//     dramaAverageRates.reduce((acc, i) => {
+//         if (i.rate != '' && i.rate != undefined){
+//             return acc + i.rate
+//         } else {
+//             return acc + 0
+//         }
+//     }, 0);
+
+//     let averageDramaRates = (dramaMovies / array.length).toFixed(2);
+//     averageDramaRates = Number(averageDramaRates); 
+//     return averageDramaRates;
+// }
+
+
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+const orderByYear = (array) => {
+
+    // let a = JSON.stringify(array)
+    // let b = JSON.parse(a)
+
+    let movieByYear = JSON.parse(JSON.stringify(array))
+    //let movieByYear2 = array
+
+
+    movieByYear.sort((a, b) => {
+        if (a.year > b.year) {
+                    return 1;
+                }
+                else if (a.year < b.year) {
+                  return -1;
+                }
+                else {
+                    if (a.title > b.title) {
+                        return 1;
+                    }
+                    else if (a.title < b.title) {
+                      return -1;
+                    }
+                    else {
+                      return 0
+                    }
+                }
+    });
+    return movieByYear;
+};
+
+
+
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
 
+const orderAlphabetically = (array) => {
+
+    let orderedMovies = JSON.parse(JSON.stringify(array))
+    let orderedTitles = orderedMovies
```

Comment:
this line does not make much sense since it is pointing to the same reference,  so using the variable orderedMovies will be exactly the same of using the variable orderedTitles. Since they both point to the same memory location.